hs/src/Reach/Debugger.hs

```diff
@@ -81,15 +80,19 @@ type Session = M.Map Id State
 type Params = String -- JSON.Value
 
 -- interpreter
-type Gas = Natural
+type Gas = Integer
 
 -- determine if program is already a value
 get_prog_val :: LLProg -> Maybe DLVal
 get_prog_val = undefined
 
 interp :: Gas -> LLProg -> App (Maybe DLVal)
```

Comment:
FWIW, this signature isn't going to work, because `LLProg`s don't return to values, they would reduce to more `LLProg`s, as you're going to need an environment to find the values. My taste re "gas" is to have a single step function and then a wrapper that calls many times, like how Redex works